### PR TITLE
Spotify user login with Spring Boots OAuth integration

### PR DESCRIPTION
**Pull request by hedgehog1833**
As discussed for google users the spotify sync does currently not work. I'm not 100% sure if the config in the application-preview.yml is necessary..

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Rework of Spotify Authorization](https://trello.com/c/K4zYIgm5/382-rework-of-spotify-authorization)

File: spotify/src/main/java/rocks/metaldetector/spotify/client/SpotifyUserLibraryClientImpl.java [hunk 1]
```diff
@@ -85,11 +77,10 @@ public SpotifyFollowedArtistsPage fetchFollowedArtists(String token, String next
     return result.getArtistsPage();
   }
 
-  private HttpEntity<Object> createHttpEntity(String token) {
+  private HttpEntity<Object> createHttpEntity() {
```
Comment (DanielW1987) [inline spotify/src/main/java/rocks/metaldetector/spotify/client/SpotifyUserLibraryClientImpl.java:80]: If the token is already set in an interceptor, can't the accept and charset be set there as well? Seems unnecessarily elaborate to me here.

File: webapp/src/main/java/rocks/metaldetector/service/user/events/UserDeletionEventListener.java [hunk 2]
```diff
@@ -42,7 +40,6 @@ public void onApplicationEvent(UserDeletionEvent event) {
     AbstractUserEntity user = event.getUserEntity();
     log.info("User '" + user.getPublicId() + "' deleted");
 
-    spotifyAuthorizationRepository.deleteByUser(user);
```
Comment (DanielW1987) [inline webapp/src/main/java/rocks/metaldetector/service/user/events/UserDeletionEventListener.java:45]: What happens to the record in the OAuth table when a user deletes himself?

File: webapp/src/main/resources/application-preview.yml [hunk 3]
```diff
@@ -21,6 +21,12 @@ spring:
           auth: true
           starttls:
             enable: true
+  security:
+    oauth2:
+      client:
+        registration:
+          spotify-app:
+            redirect-uri: "https://{baseHost}{basePort}{basePath}/settings/spotify-synchronization"
```
Comment (DanielW1987) [inline webapp/src/main/resources/application-preview.yml:29]: where is `baseHost`, `basePort` and `basePath` defined?

Comment (DanielW1987) [reply webapp/src/main/resources/application-preview.yml:29]: How do the values differ from the `application.yml`? I don't think we need them. But not sure.

**Review by DanielW1987 — COMMENTED** [review]
Looks good

File: webapp/src/main/java/rocks/metaldetector/service/user/events/UserDeletionEventListener.java (hunk 2 above)
Comment (DanielW1987) [reply webapp/src/main/java/rocks/metaldetector/service/user/events/UserDeletionEventListener.java:45]: ok, this is already fixed.

**Review by DanielW1987 — APPROVED** [review]
Maybe you can address these issues in your current work

File: spotify/src/main/java/rocks/metaldetector/spotify/client/SpotifyUserLibraryClientImpl.java (hunk 1 above)
Comment (hedgehog1833) [reply spotify/src/main/java/rocks/metaldetector/spotify/client/SpotifyUserLibraryClientImpl.java:80]: We do that in multiple clients, I created a card for it.

File: webapp/src/main/resources/application-preview.yml (hunk 3 above)
Comment (hedgehog1833) [reply webapp/src/main/resources/application-preview.yml:29]: I think somehow spring does not yet know (at least from an application.yml perspective) that ssl is activated on preview and prod. So probably we can refactor some more here, but I'll check that out later. Don't want to break more than necessary.

**Pull request merged**